tensorflow/cppbuild.sh

```diff
@@ -151,45 +151,47 @@ case $PLATFORM in
-			fi
-			
-			# two xeon core need around 2min
-			if [[ ! -f ../build/tf_c_python_api.dir/Release/tf_c_python_api.lib ]]; then
-				MSBuild.exe tf_python_protos_cc.vcxproj //p:BuildProjectReferences=false //p:Configuration=Release //p:CL_MPCount=$MAKEJ //p:Platform=x64 //p:PreferredToolArchitecture=x64 //filelogger 
-				MSBuild.exe tf_c_python_api.vcxproj //p:BuildProjectReferences=false //p:Configuration=Release //p:CL_MPCount=$MAKEJ //p:Platform=x64 //p:PreferredToolArchitecture=x64 //filelogger 
-			fi
-		fi
-			
-		# the remaining parts, if the variable is not set or set to 0
-		if [[ "${PARTIAL_CPPBUILD:-0}" == "0" ]]; then
-	
-			# two xeon core need around 3h 15min
-			if [[ ! -f ../build/Release/tensorflow_static.lib ]] && [[ "${PARTIAL_CPPBUILD:-}" != "1" ]]; then
-				MSBuild.exe tf_core_kernels.vcxproj //p:BuildProjectReferences=false //p:Configuration=Release //p:CL_MPCount=$MAKEJ //p:Platform=x64 //p:PreferredToolArchitecture=x64 //filelogger 
-				MSBuild.exe tf_tools_transform_graph_lib.vcxproj //p:BuildProjectReferences=false //p:Configuration=Release //p:CL_MPCount=$MAKEJ //p:Platform=x64 //p:PreferredToolArchitecture=x64 //filelogger 
-				MSBuild.exe tensorflow_static.vcxproj //p:BuildProjectReferences=false //p:Configuration=Release //p:CL_MPCount=$MAKEJ //p:Platform=x64 //p:PreferredToolArchitecture=x64 //filelogger
-			fi
-		fi
-		
+
+        MSBUILD_OPTIONS="//p:Configuration=Release //p:CL_MPCount=$MAKEJ //p:Platform=x64 //p:PreferredToolArchitecture=x64 //filelogger"
+
+        # partial build if the variable is not set or set to 1
+        if [[ "${PARTIAL_CPPBUILD:-1}" == "1" ]]; then
+
+            # two xeon core need around 30min
+            MSBuild.exe tf_c.vcxproj $MSBUILD_OPTIONS
+            MSBuild.exe tf_core_cpu.vcxproj $MSBUILD_OPTIONS
+            MSBuild.exe tf_cc.vcxproj $MSBUILD_OPTIONS
+            MSBuild.exe tf_core_ops.vcxproj $MSBUILD_OPTIONS
+            MSBuild.exe tf_core_distributed_runtime.vcxproj $MSBUILD_OPTIONS
+            MSBuild.exe tf_core_direct_session.vcxproj $MSBUILD_OPTIONS
+
+            # two xeon core need around 1h 40min
+            if [[ "$EXTENSION" == *gpu ]]; then
+                MSBuild.exe tf_core_kernels_cpu_only.vcxproj $MSBUILD_OPTIONS
+                MSBuild.exe tf_stream_executor.vcxproj $MSBUILD_OPTIONS
+                MSBuild.exe tf_core_gpu_kernels.vcxproj $MSBUILD_OPTIONS
+            fi
+
+            # two xeon core need around 2min
+            if [[ ! -f ../build/tf_c_python_api.dir/Release/tf_c_python_api.lib ]]; then
+                MSBuild.exe $MSBUILD_OPTIONS
+                MSBuild.exe tf_c_python_api.vcxproj $MSBUILD_OPTIONS
+            fi
+        fi
+
+        # the remaining parts, if the variable is not set or set to 0
+        if [[ "${PARTIAL_CPPBUILD:-0}" == "0" ]]; then
+
+            # two xeon core need around 3h 15min
+            if [[ ! -f ../build/Release/tensorflow_static.lib ]] && [[ "${PARTIAL_CPPBUILD:-}" != "1" ]]; then
```

Comment:
We can remove the second part of line 188 `&& [[ "${PARTIAL_CPPBUILD:-}" != "1" ]]`